test/BTCUtilsJs.test.js

```diff
@@ -0,0 +1,442 @@
+    assert.isFalse(res);
+
+    // no outputs stated
+    res = await BTCUtilsJs.validateVout('0x004897070000000000220020a4333e5612ab1a1043b25755c89b16d55184a42f81799e623e6bc39db8539c180000000000000000166a14edb1b5c2f39af0fec151732585b1049b07895211');
+    assert.isFalse(res);
+
+    // fewer bytes in vout than stated
+    res = await BTCUtilsJs.validateVout('0x024897070000000000220020a4333e5612ab1a1043b25755c89b16d55184a42f81799e623e6bc39db8539c180000000000000000166a14edb1b5c2f39af0fec151732585b1049b078952');
+    assert.isFalse(res);
+
+    // more bytes in vout than stated
+    res = await BTCUtilsJs.validateVout('0x024897070000000000220020a4333e5612ab1a1043b25755c89b16d55184a42f81799e623e6bc39db8539c180000000000000000166a14edb1b5c2f39af0fec151732585b1049b078952111111111111111');
+    assert.isFalse(res);
+  });
+
+  it('determines output length properly', async () => {
+    let res;
+    res = await BTCUtilsJs.determineOutputLength('0x00000000000000002200');
+    assert.equal(res, 43n);
+
+    res = await BTCUtilsJs.determineOutputLength('0x00000000000000001600');
+    assert.equal(res, 31n);
+
+    res = await BTCUtilsJs.determineOutputLength('0x0000000000000000206a');
+    assert.equal(res, 41n);
+
+    res = await BTCUtilsJs.determineOutputLength('0x000000000000000002');
+    assert.equal(res, 11n);
+
+    res = await BTCUtilsJs.determineOutputLength('0x000000000000000000');
+    assert.equal(res, 9n);
+
+    res = await BTCUtilsJs.determineOutputLength('0x000000000000000088');
+    assert.equal(res, 145n);
+
+    try {
+      res = await BTCUtilsJs.determineOutputLength('0x0000000000000000FF00');
+      assert(false, 'Expected an error');
+    } catch (e) {
+      assert.include(e.message, 'Multi-byte VarInts not supported');
+    }
+  });
+
+  it('extracts outputs at specified indices', async () => {
+    let res;
+    res = await BTCUtilsJs.extractOutputAtIndex(constants.OP_RETURN.VOUT, 0);
+    assert.equal(res, constants.OP_RETURN.INDEXED_OUTPUTS[0].OUTPUT);
+    res = await BTCUtilsJs.extractOutputAtIndex(constants.OP_RETURN.VOUT, 1);
+    assert.equal(res, constants.OP_RETURN.INDEXED_OUTPUTS[1].OUTPUT);
+    res = await BTCUtilsJs.extractOutputAtIndex(TWO_IN_TX_VOUT, 0);
+    assert.equal(res, '0x4db6000000000000160014455c0ea778752831d6fc25f6f8cf55dc49d335f0');
+    res = await BTCUtilsJs.extractOutputAtIndex(TWO_IN_TX_VOUT, 1);
+    assert.equal(res, '0x40420f0000000000220020aedad4518f56379ef6f1f52f2e0fed64608006b3ccaff2253d847ddc90c91922');
+  });
+
+  it('extracts a root from a header', async () => {
+    const res = await BTCUtilsJs.extractMerkleRootBE(HEADER_170);
+    assert.equal(res, '0x7dac2c5666815c17a3b36427de37bb9d2e2c5ccec3f8633eb91a4205cb4c10ff');
+  });
+
+  it('extracts the target from a header', async () => {
+    const res = await BTCUtilsJs.extractTarget(HEADER_170);
```

Comment:
@dominiquejane if this is the test that's running, you need to call `deserializeHex` here before passing the header in

Reply:
ok. Other tests also use this header. Do they all need to be deserialized first?

Reply:
Yes. It might be wise to simply deserialize the header where we declare it a constant in the header file

Reply:
in the test file*

Reply:
k